src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2018,6 +2018,214 @@ TEST_F(
+  params[strings::hmi_display_language_desired] = mobile_api::Language::EN_US;
+
+  rai[strings::params][strings::connection_key] = kConnectionKey;
+  rai[strings::msg_params][strings::app_name] = kAppName;
+  rai[strings::msg_params][strings::app_hmi_type] = "WEB_VIEW";
+  rai[strings::msg_params][strings::is_media_application] = false;
+  rai[strings::msg_params][strings::sync_msg_version][strings::minor_version] =
+      APIVersion::kAPIV2;
+  rai[strings::msg_params][strings::sync_msg_version][strings::major_version] =
+      APIVersion::kAPIV6;
+
+  rai[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  const auto rai_ptr = std::make_shared<smart_objects::SmartObject>(rai);
+
+  std::unique_ptr<plugin_manager::RPCPluginManager> rpc_plugin_manager(
+      new MockRPCPluginManager());
+  app_manager_impl_->SetPluginManager(rpc_plugin_manager);
+  auto wep_app = app_manager_impl_->RegisterApplication(rai_ptr);
+  wep_app->set_is_media_application(false);
+
+  EXPECT_EQ(protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2,
+            wep_app->protocol_version());
+  EXPECT_EQ(APIVersion::kAPIV2, wep_app->version().min_supported_api_version);
+  EXPECT_EQ(APIVersion::kAPIV6, wep_app->version().max_supported_api_version);
+
+  const WindowID main_window_id = 0;
+  // Initial HMI level, audio, video streaming states and system context
+  // set after app registration
+  EXPECT_EQ(mobile_apis::HMILevel::INVALID_ENUM,
+            wep_app->hmi_level(main_window_id));
+  EXPECT_EQ(mobile_apis::VideoStreamingState::INVALID_ENUM,
+            wep_app->video_streaming_state());
+  EXPECT_EQ(mobile_apis::AudioStreamingState::INVALID_ENUM,
+            wep_app->audio_streaming_state());
+  EXPECT_EQ(mobile_apis::SystemContext::SYSCTXT_MAIN,
+            wep_app->system_context(main_window_id));
+
+  // Const shared ptr is required for OnAppActivated due to its signature
+  ApplicationConstSharedPtr const_wep_app = wep_app;
+  EXPECT_CALL(*mock_app_service_manager_, OnAppActivated(const_wep_app));
+
+  // Activate Webengine projection non-media app
+  // to check its audio & video streaming states in FULL Hmi level
+  app_manager_impl_->ActivateApplication(wep_app);
+
+  // WEP non-media app should be always not_streamable
+  // and not_audible even in FULL HMI Level
+  EXPECT_EQ(mobile_apis::HMILevel::HMI_FULL,
+            wep_app->hmi_level(main_window_id));
+  EXPECT_EQ(mobile_apis::VideoStreamingState::NOT_STREAMABLE,
+            wep_app->video_streaming_state());
+  EXPECT_EQ(mobile_apis::AudioStreamingState::NOT_AUDIBLE,
+            wep_app->audio_streaming_state());
+  EXPECT_EQ(mobile_apis::SystemContext::SYSCTXT_MAIN,
+            wep_app->system_context(main_window_id));
+
+  // Deactivate Webengine projection non-media app
+  // to check its new HMI level and audio & video streaming states
+  app_manager_impl_->state_controller().DeactivateApp(wep_app, main_window_id);
```

Comment:
@AByzhynar Is it possible to split this test into several separate tests?

Reply:
@AByzhynar the tests could be splitted into 4 parts to check Register, Activate, Deactivate and  Exit separately. Also message construction and state checks could be moved to separate functions.

Reply:
@mvorobio Activate, Deactivate and Exit separately have no sense without Registration? Make an example how to fo it without registration.

Reply:
@AByzhynar  you're right, Activate, Deactivate and Exit tests should contain registration as well but without additional checks. I think, the tests would be clearer.

Reply:
@mvorobio Thank you. The functions separation was done in  e570bb7a3d5d48b9e64a5172d70804c532ec880f